sammba/registration/struct.py

```diff
@@ -9,9 +9,10 @@
 def anats_to_common(anat_filenames, write_dir, brain_volume,
                     registration_kind='affine',
                     nonlinear_levels=[1, 2, 3],
-                    nonlinear_minimal_patch=75,
-                    convergence=0.005, caching=False, verbose=False,
-                    unifize_kwargs=None, brain_extraction_unifize_kwargs=None):
+                    nonlinear_minimal_patches=[75],
+                    nonlinear_weight=None,
```

Comment:
better to use nonlinear_weight_filename I think

Reply:
on two grounds it shouldn't be: 1) in the future this could also be a function passed on to modify the automatically-generated mask, 2) many/all uses of "-filename(s)" in input variables should gradually be removed as they unnecessarily lengthen the names of variables that are inherently/obviously files, and are inconsistent with the format of the flags eventually used with the underlying AFNI commands

Reply:
> 2) many/all uses of "-filename(s)" in input variables should gradually be removed as they unnecessarily lengthen the names of variables

...filename can be replaced with just _file
>that are inherently/obviously files,

this is obvious to us. It can be a folder, it can be a nibabel image, it can be everything. In the specific case of weight I already misunderstood this type and changed some code (remember we talked about that) to make it take integers and get this code merged ! So no, it is not obvious at all that a weight is a file

>and are inconsistent with the format of the flags eventually used with the underlying AFNI commands

This is not very important, cause end users are not assumed to guess which AFNI function we are using and what are its parameters. nipype wrapping itself differs for many parameters from the AFNI parameter names and more generally from the wrapped interfaces names. What is important is the overall consistency of the tool: if it's called filename here, it must be called filename everywhere

Reply:
Done